test/convergence.jl

```diff
@@ -75,18 +75,16 @@ function tabulate_convergence_orders_imex_ark()
     ]
     tabs = map(t -> t(), tabs)
     test_cases = all_test_cases(Float64)
-    results = convergence_order_results(tabs, test_cases)
-    algs = algorithm.(tabs)
-    prob_names = map(t -> t.test_name, test_cases)
-    expected_orders = ODE.alg_order.(tabs)
-    tabulate_convergence_orders(prob_names, algs, results, expected_orders; tabs)
+    get_alg = (tab, test_case) -> IMEXARKAlgorithm(tab, NewtonsMethod(; max_iters = test_case.linear_implicit ? 1 : 2))
+    results = convergence_order_results(tabs, test_cases, get_alg)
+    tabulate_convergence_orders(results)
     return results
 end
 tabulate_convergence_orders_imex_ark()
 
 function tabulate_convergence_orders_ark()
     # IMEX
-    co = Dict()
+    co = Dict(:auto => Dict(), :nonauto => Dict())
```

Comment:
Why split this into auto and non-auto? I'd like to combine all of the tables, and make it easy to extend to more problems, this seems it restricts us to these specific problems

Reply:
It was challenging to convert `results` to a 2D array in its original format (now that the `algs` are not generated outside of `tabulate_convergence_orders`); changing it to a `Dict` of `Dict`s makes it significantly easier.

Reply:
> now that the algs are not generated outside of tabulate_convergence_orders

I don't want to make this change, I want `algs` to be passed into `tabulate_convergence_orders`.

Reply:
Okay, I really don't want to get bogged down in this new testing framework. If you don't like this PR, could you please propose an alternative PR that achieves the same purpose? I'd like to move on to testing IMEX SSPRK with the limiter test case.

Reply:
Agreed. I think I prefer we close the PR.

> could you please propose an alternative PR that achieves the same purpose

This PR isn't changing any source code, so it's not really blocking us from doing anything. We can always add additional tests that test things differently.